web/src/testing/render.tsx

```diff
@@ -51,7 +52,7 @@ export const renderRoute = (ui: ReactElement, options: RenderRouteOptions): Exte
     },
   ]
   const router = createMemoryRouter(routes, {
-    initialEntries: [{ pathname: options.pathname, search: options.searchParams }],
+    initialEntries: options.initialEntries ?? [{ pathname: options.pathname, search: options.searchParams }],
```

Comment:
🙃 How about
```suggestion
    initialEntries: [...options.previousRoutes ?? [], { pathname: options.pathname, search: options.searchParams }],
```
That way, there is no need to also pass the current route :) Also, I think `initalRoutes` or even `previousRoutes` would be a little more descriptive.